crypto/dh/dh_ameth.c

```diff
@@ -536,6 +536,62 @@ static int dhx_pkey_import_from(const OSSL_PARAM params[], void *vpctx)
     return dh_pkey_import_from_type(params, vpctx, EVP_PKEY_DHX);
 }
 
+#define dh_bn_dup_check(bn) \
+{ \
+  const BIGNUM *f = DH_get0_##bn(dh); \
+  \
+  if (f != NULL && (dupkey->bn = BN_dup(f)) == NULL) \
+      goto err; \
```

Comment:
I'm pretty sure the code style says no flow control in macros.
This only appears to be used twice, so not such a pain to inline here.

There are other flavours throughout.

Reply:
I'll change this to use a function instead of a macro, although I believe we have other cases where flow control appears in macros.

Reply:
I'm sure we do, the coding style doesn't seem to be static.
I don't have a problem with changing these in a separate PR.